File: cplus/leetcode/202608/algo3781.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    long long maximumScore(vector<int>& nums, string s) {
        long long ans = 0; int n = nums.size();
        priority_queue<int> pq;

        for (int i=0; i<n; i++) {
            pq.emplace(nums[i]);
            if (s[i] == '1') {
                ans += pq.top(); pq.pop();
            }
        }
        return ans;
    }
};

File: python/algo/202608/algo3781.py
import heapq
from typing import List

class Solution:
    def maximumScore(self, nums: List[int], s: str) -> int:
        pq, ans = [], 0
        for v, ch in zip(nums, s):
            heapq.heappush(pq, -v)
            if ch == '1':
                ans -= heapq.heappop(pq)
        return ans